src/components/About/Contact/Contact.react.js

```diff
@@ -96,6 +139,23 @@ const Contact = props => {
 Contact.propTypes = {
   history: PropTypes.object,
   location: PropTypes.object,
+  google: PropTypes.object,
+  mapKey: PropTypes.string,
 };
 
-export default Contact;
+function mapStateToProps(store) {
+  return {
+    mapKey: store.app.apiKeys.mapKey || '',
+  };
+}
+
+const LoadingContainer = props => <CircularLoader height={27} />;
+
+export default withRouter(
+  connect(mapStateToProps)(
```

Comment:
Can you also abstract away this logic. Maybe by making a HOC for this, as it is also getting redundant.

Reply:
Sir, there were several problems during implementation of hoc .I tried several times but HOC returned null every-time .  It was because the googleAPI-wrapper has no other way to implement it over a component 
 
> GoogleApiWrapper(props => ({apiKey: props.mapKey}))(Component)

if we implement it using HOC we would have to import state from redux store and then pass mapKey as a prop to it ,But for most of the components require binding with redux store too.So, multiple bindings with redux store would apply in such cases for which we have to pass down the props to component to be rendered and I don't think it would be a good choice